include/swift/AST/Decl.h

```diff
@@ -558,10 +558,12 @@ class alignas(1 << DeclAlignInBits) Decl {
     IsIncompatibleWithWeakReferences : 1
   );
 
-  SWIFT_INLINE_BITFIELD(StructDecl, NominalTypeDecl, 1,
+  SWIFT_INLINE_BITFIELD(StructDecl, NominalTypeDecl, 1+1,
     /// True if this struct has storage for fields that aren't accessible in
     /// Swift.
-    HasUnreferenceableStorage : 1
+    HasUnreferenceableStorage : 1,
+    /// True if this struct is imported from C++ and not trivially copyable.
+    HasNonTrivialCopyOrMove : 1
```

Comment:
Removing the negation seems like it would simplify the semantics? I.e. `TriviallyCopyable`, or even `CxxTriviallyCopyable`. Using the "trivially copyable" wording (which is a C++ term) would also emphasize that this is exactly the C++ concept.

Reply:
I considered that for exactly the reasons that you mention. I went with the current semantics anyway, because it is not _only_ the C++ concept. It is "imported *AND* not trivially copyable". If I invert the semantics it becomes "not imported or trivially copyable", which is a mouthful and still not a nice concept. I could go with "trivially copyable" alone, but then what does that mean for the non-C++ case? Or with "imported and trivially copyable", but then I need to check "imported" when I use the bit...

Also the concepts combine in a nicer way: if either of those bits is set it isn't loadable any more...

Reply:
> It is "imported _AND_ not trivially copyable".

Hm... good point. I have to admit though that the "is imported" bit wasn't really clear to me when I wrote the comment above... does it make sense to have this be part of the name?

Also, maybe we can still keep the C++ term "trivially copyable" and say "not trivially copyable"?

In sum, this would give us maybe `CxxNotTriviallyCopyable` or `ImportedNotTriviallyCopyable`?

Reply:
I changed the name to `IsCxxNotTriviallyCopyable`.

Reply:
SG